File: typescript-nextjs/src/service.ts
import { NotImplementedError, ServiceError } from "./errors";
import { Event, EventInput } from "./models";
import { MemoryStore } from "./store";

export class EventService {
  private store: MemoryStore;

  constructor(store: MemoryStore) {
    this.store = store;
  }

  create(data: EventInput): Event {
    throw new NotImplementedError();
  }

  read(eventId: string): Event {
    throw new NotImplementedError();
  }

File: c/src/service.c
#include "service.h"

ServiceStatus service_create(MemoryStore *store, const EventInput *data, const Event **out) {
    return SERVICE_NOT_IMPLEMENTED;
}

ServiceStatus service_read(MemoryStore *store, const char *event_id, const Event **out) {
    return SERVICE_NOT_IMPLEMENTED;
}

ServiceStatus service_update(MemoryStore *store, const char *event_id, const EventInput *data,
                            const Event **out) {
    return SERVICE_NOT_IMPLEMENTED;
}

ServiceStatus service_delete(MemoryStore *store, const char *event_id) {
    return SERVICE_NOT_IMPLEMENTED;
}
